packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -78,7 +83,8 @@ export class Span implements api.Span, ReadableSpan {
     this.links = links;
     this.startTime = timeInputToHrTime(startTime);
     this.resource = parentTracer.resource;
-    this.instrumentationLibrary = parentTracer.instrumentationLibrary;
+    this.instrumentationLibrary = parentTracer.instrumentationScope as InstrumentationLibrary;
```

Comment:
There should be no cast needed as the interfaces are equal. If they differ a conversion would be needed.
An alternative would be to change ` instrumentationLibrary` to a getter returing `this.instrumentationScope`.